tools/src/test/java/org/apache/kafka/tools/consumer/group/DescribeConsumerGroupTest.java

```diff
@@ -0,0 +1,855 @@
+            ConsumerGroupExecutor executor = addConsumerGroupExecutor(1, TOPIC, group, groupProtocol);
+            List<String> cgcArgs = new ArrayList<>(Arrays.asList("--bootstrap-server", bootstrapServers(listenerName()), "--describe", "--group", group));
+            cgcArgs.addAll(Arrays.asList(describeType));
+            ConsumerGroupCommand.ConsumerGroupService service = getConsumerGroupService(cgcArgs.toArray(new String[0]));
+
+            TestUtils.waitForCondition(() -> {
+                scala.Tuple2<String, String> res = kafka.utils.TestUtils.grabConsoleOutputAndError(describeGroups(service));
+                return res._1.trim().split("\n").length == 2 && res._2.isEmpty();
+            }, "Expected describe group results with one data row for describe type '" + String.join(" ", describeType) + "'");
+
+            // stop the consumer so the group has no active member anymore
+            executor.shutdown();
+            TestUtils.waitForCondition(
+                () -> kafka.utils.TestUtils.grabConsoleError(describeGroups(service)).contains("Consumer group '" + group + "' has no active members."),
+                "Expected no active member in describe group results with describe type " + String.join(" ", describeType));
+        }
+    }
+
+    @ParameterizedTest(name = TEST_WITH_PARAMETERIZED_QUORUM_AND_GROUP_PROTOCOL_NAMES)
+    @MethodSource({"getTestQuorumAndGroupProtocolParametersAll"})
+    public void testDescribeOffsetsOfExistingGroupWithNoMembers(String quorum, String groupProtocol) throws Exception {
+        createOffsetsTopic(listenerName(), new Properties());
+
+        // run one consumer in the group consuming from a single-partition topic
+        ConsumerGroupExecutor executor = addConsumerGroupExecutor(1, TOPIC, GROUP, RangeAssignor.class.getName(), Optional.empty(), Optional.empty(), true, groupProtocol);
+
+        String[] cgcArgs = new String[]{"--bootstrap-server", bootstrapServers(listenerName()), "--describe", "--group", GROUP};
+        ConsumerGroupCommand.ConsumerGroupService service = getConsumerGroupService(cgcArgs);
+
+        TestUtils.waitForCondition(() -> {
+            scala.Tuple2<Option<String>, Option<Seq<ConsumerGroupCommand.PartitionAssignmentState>>> res = service.collectGroupOffsets(GROUP);
+            return res._1.map(s -> s.contains("Stable")).getOrElse(() -> false)
+                && res._2.map(c -> c.exists(assignment -> Objects.equals(assignment.group(), GROUP) && assignment.offset().isDefined())).getOrElse(() -> false);
+        }, "Expected the group to initially become stable, and to find group in assignments after initial offset commit.");
+
+        // stop the consumer so the group has no active member anymore
+        executor.shutdown();
+
+        Tuple2<scala.Tuple2<Option<String>, Option<Seq<ConsumerGroupCommand.PartitionAssignmentState>>>, Boolean> res = ToolsTestUtils.computeUntilTrue(
```

Comment:
We can leverage `TestUtils.waitForCondition` here. The developers can print more information manually if they need to dig in something in the future.
```java
        TestUtils.waitForCondition(() -> {
            scala.Tuple2<Option<String>, Option<Seq<ConsumerGroupCommand.PartitionAssignmentState>>> offsets = service.collectGroupOffsets(GROUP);
            Option<String> state = offsets._1;
            Option<Seq<ConsumerGroupCommand.PartitionAssignmentState>> assignments = offsets._2;
            @SuppressWarnings("unchecked")
            Seq<ConsumerGroupCommand.PartitionAssignmentState> testGroupAssignments = assignments.get().filter(a -> Objects.equals(a.group(), GROUP)).toSeq();
            ConsumerGroupCommand.PartitionAssignmentState assignment = testGroupAssignments.head();
            return state.map(s -> s.contains("Empty")).getOrElse(() -> false) &&
                    testGroupAssignments.size() == 1 &&
                    assignment.consumerId().map(c -> c.trim().equals(ConsumerGroupCommand.MISSING_COLUMN_VALUE())).getOrElse(() -> false) && // the member should be gone
                    assignment.clientId().map(c -> c.trim().equals(ConsumerGroupCommand.MISSING_COLUMN_VALUE())).getOrElse(() -> false) &&
                    assignment.host().map(c -> c.trim().equals(ConsumerGroupCommand.MISSING_COLUMN_VALUE())).getOrElse(() -> false);
        }, "failed to collect group offsets");
```

Reply:
Done.